serde/src/main/java/com/amazon/ionhiveserde/formats/IonOutputFormat.java

```diff
@@ -76,15 +80,27 @@
                                                              final Progressable progress)
         throws IOException {
 
-        final FileSystem fs = finalOutPath.getFileSystem(jc);
-        final OutputStream out = fs.create(finalOutPath, progress);
-
-        // If we are passed in a reporter, make sure we call incrCounters during write
-        Optional<Reporter> reporter = Optional.empty();
-        if (progress instanceof Reporter) {
-            reporter = Optional.of((Reporter) progress);
+        // I believe we should expect the following to hold:
+        // isCompressed == FileOutputFormat.getCompressOutput(jc);
```

Comment:
In recent testing wasn't there some confusion about which parameter governed this in practice? Whether it was e.g. [`mapreduce.output.fileoutputformat.compress`](https://github.com/c9n/hadoop/blob/master/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/output/FileOutputFormat.java#L57) or `hive.exec.compress.output`?

Maybe this is helpful: https://www.edureka.co/community/34339/difference-compress-mapreduce-fileoutputformat-compress ?

In any case we need to know what to tell users here, let's figure out what the deal is. I suppose an easier option might be to just delete this comment- this is governed at a level above us.

But it does make me wonder whether `FileOutputFormat.COMPRESS_CODEC` is the value we ought to be paying attention to or whether there's some analogy to `hive.exec.compress.output`.

Reply:
It kind of looks like 3 properties need to be set to compress? 

https://docs.cloudera.com/HDPDocuments/Ambari-1.6.1.0/bk_ambari_reference/content/ambari-ref-lzo-hive-queries.html
https://www.ibm.com/docs/en/spectrum-scale-bda?topic=hive-tuning

Reply:
Looked into references and tested a bit, `hive.exec.compress.output=true;` applies to any output that will be stored on HDFS while `mapreduce.output.fileoutputformat.compress` only applies to the MapReduce Job output. In our case (E.g, EMR), `hive.exec.compress.output=true` is necessary but `mapreduce.output.fileoutputformat.compress` is not.

For example,
```
SET hive.exec.compress.output=true; 
SET mapreduce.output.fileoutputformat.compress.codec=gzip;
``` 
above would output a compressed file,
```
SET mapreduce.output.fileoutputformat.compress=true;
SET mapreduce.output.fileoutputformat.compress.codec=gzip;
``` 
but above does not.